### DISCUSSION TITLE
Is this preformace normal or something wrong on my set up

### DISCUSSION THREAD

**Discussion by dexterhuh** [oobabooga/text-generation-webui#1370] [General]
I got an rtx3060 gpu (6GB VRAM) 32GB RAM and it takes like 15 seconds to finish replying "Hi". I'm using vicuna-13b-4bit-128g.safetensors.

For comparing the speed I tried llama.ccp, the CPU version (I got CPU AMD 3090 with 32GB RAM), and 3 seconds to reply "Hi". A way faster I feel. The model is Vicuna 13B as well.

When I say slow I mean the words show up one by one is slow.

Is this normal?

**Comment by DanielusG** [comment]
Depending on which model you load, if it is too big for your GPU it will be put into your RAM and will be slow.

**Reply by dexterhuh** [reply]
@DanielusG thank you for your reply, I have updated my text.

GPU version  rtx3060 gpu (6GB VRAM) 32GB RAM
llama.ccp, the CPU version  (I got CPU AMD 3090 with 32GB RAM)

Both are Vicuna 13B

**Reply by DanielusG** [reply]
it is normal, Vicuna 13B takes more than 6GB of VRAM, so you are using your RAM

**Comment by MaiJames** [comment]
The requirements for LLaMA 13B 4bit quantized on GPU are 10GB of VRAM & 12GB of RAM/Swap to load the model.
The requirement for the LLaMA 13B 4bit quantized on llama.cpp, running on CPU, is 7,8GB of RAM.

From my testing, the Vicuna 13B 4bit 128g, based on the LLaMA models, takes almost 12GB of VRAM on inference.

So I guess when you are using the text generation webui, maybe you use the flag `--auto-devices`, and the model is split across the VRAM and the RAM, the transfer speed between the two bottlenecks the inference speed, and it gets really slow as a result.

You could try to run all the model on CPU with the text generation webui, using the corresponding flags, or maybe try with another model.
Although from my testing, the models with the lowest VRAM requirements (Alpaca Native 7B 4bit, LLaMa 7B 4bit & Pygmalion 6B 4bit) needed at least 7GB, so you'll have the same problem.

I think with your current configuration, you'll get the best speeds on CPU for now.
